projects/week_2/Instructions.md

```diff
@@ -0,0 +1,275 @@
+
+For this project, we'll be working in the [obstacle_detector](../../obstacle_detector) package. The starter code for this project has defined a node for you. This node, `ObstacleDetector`, will subscribe to an image topic. Each image that comes across that topic will need to be filtered to highlight the pixels that match our target color. You'll then re-project the image with a homography matrix we provide to turn the image into a top-down map that the node will publish as an occupancy grid message.
+
+## 2 Running this project
+
+You can run this week's project using the week 2 launch file. This will startup the simulator, rviz, and your obstacle detection code.
+
+```bash
+$ ros2 launch rj_training_bringup week_2.launch.xml
+```
+
+To move the robot around and test your code at different positions, use the same joystick or keyboard control launch files as you used in week 1.
+
+If you run it now, you'll see rviz complain that no messages are being published on our occupancy grid topic.
+
+![Screenshot of rviz showing warning message](RvizNoGrid.png)
+
+Once you've got your code working, you should see a map show up in rviz's 3D viewport. The black areas are marked obstacles and should match the shape of the red rocks on the challenge mat. The white areas are free of obstacles. The grey area is the part of the reprojected image that we had no original pixel data for, so the robot doesn't know what's in that area.
+
+![Screenshot of rviz showing working obstacle detection](WorkingDetection.png)
+
+## 3 Instructions
+
+### 3.1 Create new files
+
+You'll be implementing the two steps of this project, color detection and re-projection, as new fuctions defined in separate files from the node's main file. The first thing we need to do is add these files to our package.
+
+Go to the ["src" folder](../../obstacle_detector/src), in the obstacle_detector package. Add two new files: "student_functions.cpp" and "student_functions.hpp".
+
+In the implementation file (student_functions.cpp), we need to include our new header file. Add an include statement at the top of the file to include the header file we just created.
+
+<details>
+<summary><b>Hint:</b> How to include a header</summary>
+<p>To include a header file named "my_header.hpp", we'd write the code below. Note that the use of quotes ("") instead of angle brackets (<>) indicates that this header is part of our current package.</p>
+<code>#include "my_header.hpp"</code>
+</details>
+
+In our new header file (student_functions.hpp), we need to add header guards. Add the three lines that define the header guards for this file. Remember that header guards look like this:
+
+```c++
```

Comment:
We should check with teams. IDK if they are using this or #pragma once

Reply:
The teams use a mix. I can add `#pragma once` to the recitation guide. All of the training code and videos sticks to guards.

Reply:
We can leave that to the teams then